shell/platform/android/io/flutter/embedding/android/FlutterActivityLaunchConfigs.java

```diff
@@ -18,6 +18,9 @@
       "io.flutter.embedding.android.NormalTheme";
   /* package */ static final String HANDLE_DEEPLINKING_META_DATA_KEY =
       "flutter_deeplinking_enabled";
+  /* package */ static final String USE_LEGACY_SPLASH_SCREEN_META_DATA_KEY =
+      "io.flutter.embedding.android.use-legacy-splash-screen";
```

Comment:
I'm not sure if this meta data is necessary. 

It might be worth looking at the case when the user didn't supply a custom drawable. In this case, there's no drawable, so the splash screen is unnecessary. 

https://github.com/flutter/engine/blob/f05cb777db7e91912fc6448815b4d2bd322663d3/shell/platform/android/io/flutter/embedding/android/FlutterActivity.java#L496

I was thinking that it could be the way to decide whether we install the predraw listener or not.

Reply:
This will mean that the new behavior with preDraw will not be used by default, and is instead opt-in, which sounds a bit simpler, thanks 🙂 . I'll update the PR

Reply:
do we still need `io.flutter.embedding.android.use-legacy-splash-screen` or can it be removed?

Reply:
Whoops, removed it!